### PR TITLE
[XL] LP: Complete OAUTH, reroute user appropriately, implement create account flow

### PR DESCRIPTION
**Pull request by justcord10**
user can now login using google OAuth. the user is then routed appropriately using the react router to either the account creation page or back to the landing page depending if they have an account on the DB already.  The general flow is as follows user clicks on sign in button on the landing page => user routed to the sign in page => user routed to the google login OAuth screen => user then routed to the AuthCallbackPage where the database is checked for the existing account on the accounts table, and the store is updated with the user info received from the login attempt. => if the user exists on the db they are then routed to the homepage else if the user does not exist they are routed to the account creation page where they can select a role and fill out the associated form, the user can go back at this point to change roles and some fields are prepopulated with data. a post request is kicked off to add the account data to the database. 
User Clicks of sign in:
image
User can sign in using google:
image
Google Oauth Page:
image
Automatically routed back to the home page if the account exists:
image
Automatically routed to the account creation page if the account !exists:
image
Valet creation page:
image
Driver creation page:
image

Small notes and changes, added a slice and set function on the accountsSlice.js file
Adjusted the Regex for the Phone number to better align with the phone number on the database.
on account creation certain fields are prepopulated with data depending on what is returned by google.
deleted the old supabase utility function and implemented a use supabase custom hook to integrate with the app more (thanks for all the help on this Steven, I can sleep at 2am instead of 7am haha)
added a supabase on auth state change event listener to automatically add a users session info to the store on auth change, also stores session info within the browsers local storage so that session persistance can be established. 
I think theres more but i am sleepy, that should be most of it and ill add what any more when i wake up. 
Thanks to Bruce and Steven for pounding their heads against a wall alongside me to get some of the kinks ironed out!
And Thanks to Jose for setting up the initial Google Oauth and reroutes along with the forms to make an easy integration!

### CODE REVIEW

File: src/components/login/AccountForm.jsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ const AccountForm = ({ formData, handleChange }) => {
       </label>
       <label>
         Phone Number:
-        <input type="tel" name="phoneNumber" value={formData.phoneNumber} onChange={handleChange} placeholder="123-456-7890" pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}" required />
+        <input type="tel" name="phoneNumber" value={formData.phoneNumber} onChange={handleChange} placeholder="(123) 456-7890" pattern="\([0-9]{3}\) [0-9]{3}-[0-9]{4}" required />
```
Comment (jose-m-f) [inline src/components/login/AccountForm.jsx:18]: Nice format

File: src/components/login/GarageForm.jsx [hunk 2]
```diff
@@ -1,5 +1,7 @@
+import { states } from '../../lib/states';
+
 const GarageForm = ({ formData, handleChange }) => {
-  const states = ['AL','AK','AZ','AR','CA','CO','CT','DE','FL','GA','HI','ID','IL','IN','IA','KS','KY','LA','ME','MD','MA','MI','MN','MS','MO','MT','NE','NV','NH','NJ','NM','NY','NC','ND','OH','OK','OR','PA','RI','SC','SD','TN','TX','UT','VT','VA','WA','WV','WI','WY']
```
Comment (jose-m-f) [inline src/components/login/GarageForm.jsx:2]: Good idea to use the library

File: src/components/login/GarageForm.jsx [hunk 3]
```diff
@@ -70,7 +72,7 @@ const GarageForm = ({ formData, handleChange }) => {
       </label>
       <label>
         Parking Spots:
-        <input type="number" name="garageParkingSpots" value={formData.parkingSpots} onChange={handleChange} placeholder="145" required />
+        <input type="number" name="garageParkingSpots" value={formData.parkingSpots} onChange={handleChange} placeholder="145" min={1} required />
```
Comment (jose-m-f) [inline src/components/login/GarageForm.jsx:75]: Nice data validation

**Review by jose-m-f — APPROVED** [review]
Everything looks great!

**Review by bzw24 — APPROVED** [review]
Okay with the changes, we discussed them together already.

**Pull request merged**